merkle/compact/tree_test.go

```diff
@@ -69,98 +69,49 @@ func TestAddingLeaves(t *testing.T) {
-	// api-visible calculation of root & size.
-	{
-		// First tree, add nodes one-by-one
-		tree := NewTree(rfc6962.DefaultHasher)
-		if got, want := tree.Size(), int64(0); got != want {
-			t.Errorf("Size()=%d, want %d", got, want)
-		}
-		if got, want := tree.CurrentRoot(), testonly.EmptyMerkleTreeRootHash(); !bytes.Equal(got, want) {
-			t.Errorf("CurrentRoot()=%x, want %x", got, want)
-		}
-
-		for i := 0; i < 8; i++ {
-			tree.AddLeaf(inputs[i], func(int, int64, []byte) error {
-				return nil
-			})
-			if err := checkUnusedNodesInvariant(tree); err != nil {
-				t.Fatalf("UnusedNodesInvariant check failed: %v", err)
-			}
-			if got, want := tree.Size(), int64(i+1); got != want {
-				t.Errorf("Size()=%d, want %d", got, want)
-			}
-			if got, want := tree.CurrentRoot(), roots[i]; !bytes.Equal(got, want) {
-				t.Errorf("CurrentRoot()=%v, want %v", got, want)
-			}
-			if diff := pretty.Compare(tree.Hashes(), hashes[i]); diff != "" {
-				t.Errorf("post-Hashes() diff:\n%v", diff)
-			}
-		}
-	}
-
-	{
-		// Second tree, add nodes all at once
-		tree := NewTree(rfc6962.DefaultHasher)
-		for i := 0; i < 8; i++ {
-			tree.AddLeaf(inputs[i], func(int, int64, []byte) error {
-				return nil
-			})
-			if err := checkUnusedNodesInvariant(tree); err != nil {
-				t.Fatalf("UnusedNodesInvariant check failed: %v", err)
-			}
-		}
-		if got, want := tree.Size(), int64(8); got != want {
-			t.Errorf("Size()=%d, want %d", got, want)
-		}
-		if got, want := tree.CurrentRoot(), roots[7]; !bytes.Equal(got, want) {
-			t.Errorf("CurrentRoot()=%v, want %v", got, want)
-		}
-		if diff := pretty.Compare(tree.Hashes(), hashes[7]); diff != "" {
-			t.Errorf("post-Hashes() diff:\n%v", diff)
-		}
-	}
-
-	{
-		// Third tree, add nodes in two chunks
-		tree := NewTree(rfc6962.DefaultHasher)
-		for i := 0; i < 3; i++ {
-			tree.AddLeaf(inputs[i], func(int, int64, []byte) error {
-				return nil
-			})
-			if err := checkUnusedNodesInvariant(tree); err != nil {
-				t.Fatalf("UnusedNodesInvariant check failed: %v", err)
-			}
-		}
-		if got, want := tree.Size(), int64(3); got != want {
-			t.Errorf("Size()=%d, want %d", got, want)
-		}
-		if got, want := tree.CurrentRoot(), roots[2]; !bytes.Equal(got, want) {
-			t.Errorf("CurrentRoot()=%v, want %v", got, want)
-		}
-		if diff := pretty.Compare(tree.Hashes(), hashes[2]); diff != "" {
-			t.Errorf("post-Hashes() diff:\n%v", diff)
-		}
-
-		for i := 3; i < 8; i++ {
-			tree.AddLeaf(inputs[i], func(int, int64, []byte) error {
-				return nil
-			})
-			if err := checkUnusedNodesInvariant(tree); err != nil {
-				t.Fatalf("UnusedNodesInvariant check failed: %v", err)
+	// Test the "same" thing in different ways, to ensure than any lazy update
+	// strategy being employed by the implementation doesn't affect the
+	// API-visible calculation of root & size.
+	for _, tc := range []struct {
+		desc   string
+		breaks []int
+	}{
+		{desc: "one-by-one", breaks: []int{0, 1, 2, 3, 4, 5, 6, 7, 8}},
```

Comment:
is []int required?

Reply:
Yes, otherwise the compiler says:
`./tree_test.go:79:32: missing type in composite literal`

Reply:
It is annoying as it must know the types involved.

Reply:
I feel the same way.